File: example-nonblocking/src/main/java/com/davin111/examplenonblocking/client/AccountClient.java
package com.davin111.examplenonblocking.client;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.web.reactive.function.client.WebClient;
import reactor.core.publisher.Mono;

import java.time.Duration;

@Service
@RequiredArgsConstructor
public class AccountClient {

    private WebClient.Builder webClientBuilder;

//    private WebClient webClient = webClientBuilder.build();

    public Mono<AccountResponse> getAccountResponse(Long userId) {
        return Mono.just(new AccountResponse(true)).delayElement(Duration.ofSeconds(3));
    }
}

File: example-nonblocking-kotlin/src/main/kotlin/com/davin111/examplenonblockingkotlin/client/AccountClient.kt
package com.davin111.examplenonblockingkotlin.client

import kotlinx.coroutines.delay
import org.springframework.stereotype.Service
import org.springframework.web.reactive.function.client.WebClient

@Service
class AccountClient(
    private val webClientBuilder: WebClient.Builder,
) {

    suspend fun getAccountResponse(userId: Long?): AccountResponse {
        delay(3000)
        return AccountResponse(true)
    }
}
